lib/elixir/lib/module.ex

```diff
@@ -1098,13 +1098,15 @@ defmodule Module do
 
   ## Helpers
 
-  defp preprocess_attribute(key, value) when key in [:moduledoc, :typedoc, :doc] do
-    unless match?({line, _} when is_integer(line), value) do
-      raise ArgumentError, "expected #{key} attribute given in the {line, doc} format, got: #{inspect(value)}"
-    end
+  defp preprocess_attribute(key, {line, doc} = value)
```

Comment:
Yes, non-tuple values will raise a `FunctionClauseError` right now, I did this intentionally to mimic the behaviour of `Module.add_doc/6`. I can use a `case` to refactor things a bit, but I will stand strong against the `match?/2`, would become really hard to read here 😄